src/v/kafka/server/fetch_metadata_cache.h

```diff
@@ -54,6 +57,19 @@ class fetch_metadata_cache {
         auto& e = _cache[ktp];
         e.reset_ts();
         if (offset_count > 0) {
+            if (e.bytes_per_offset.has_samples()) {
+                auto avg_bytes_per_offset = static_cast<double>(
+                  e.bytes_per_offset.get());
+                auto bytes_per_offset = static_cast<double>(total_size_bytes)
+                                        / static_cast<double>(offset_count);
+                auto abs_err = std::max(bytes_per_offset, avg_bytes_per_offset)
```

Comment:
This is an interesting way to calculate this, did you intentionally choose this over `abs(bytes_per_offset - avg_bytes_per_offset)`?